src/main.rs

```diff
@@ -79,6 +91,47 @@ struct BinOpt {
     /// The file to read C source from. \"-\" means stdin (use ./- to read a file called '-').
     /// Only one file at a time is currently accepted. [default: -]
     filename: PathBuf,
+    /// Whether or not to use color
+    color: ColorChoice,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum ColorChoice {
+    Always,
+    Auto,
+    Never,
+}
+
+impl ColorChoice {
+    fn use_color_for(self, stream: atty::Stream) -> bool {
+        match self {
+            ColorChoice::Always => true,
+            ColorChoice::Never => false,
+            ColorChoice::Auto => atty::is(stream),
+        }
+    }
+}
+
+impl std::str::FromStr for ColorChoice {
+    type Err = &'static str;
+    fn from_str(s: &str) -> Result<ColorChoice, &'static str> {
+        match s {
+            "always" => Ok(ColorChoice::Always),
+            "auto" => Ok(ColorChoice::Auto),
+            "never" => Ok(ColorChoice::Never),
+            _ => Err("Invalid color choice"),
+        }
+    }
+}
+
+impl Into<color_backtrace::termcolor::ColorChoice> for ColorChoice {
+    fn into(self) -> termcolor::ColorChoice {
+        match self {
+            ColorChoice::Always => termcolor::ColorChoice::Always,
+            ColorChoice::Auto => termcolor::ColorChoice::Auto,
+            ColorChoice::Never => termcolor::ColorChoice::Never,
```

Comment:
Personally I would `use termcolor::ColorChoice::*;` here, but it's up to you.

Reply:
I usually prefer having paths